src/container/container_iv.c

```diff
@@ -351,26 +361,39 @@ cont_iv_ent_update(struct ds_iv_entry *entry, struct ds_iv_key *key,
 	int			rc;
 
 	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
-	if (entry->iv_class->iv_class_id == IV_CONT_CAPA) {
-		rc = cont_iv_capa_ent_update(entry, key, src, priv);
-		if (rc)
-			return rc;
-	} else if (entry->iv_class->iv_class_id == IV_CONT_SNAP) {
-		struct cont_iv_entry *civ_ent = src->sg_iovs[0].iov_buf;
-
-		rc = ds_cont_tgt_snapshots_update(entry->ns->iv_pool_uuid,
-						  civ_key->cont_uuid,
-						  &civ_ent->iv_snap.snaps[0],
-						  civ_ent->iv_snap.snap_cnt);
-		if (rc)
-			return rc;
+	/* If src == NULL, it is invalidate */
+	if (src != NULL) {
+		if (entry->iv_class->iv_class_id == IV_CONT_CAPA) {
+			rc = cont_iv_capa_ent_update(entry, key, src, priv);
+			if (rc)
+				return rc;
+		} else if (entry->iv_class->iv_class_id == IV_CONT_SNAP) {
+			struct cont_iv_entry *civ_ent = src->sg_iovs[0].iov_buf;
+
+			rc = ds_cont_tgt_snapshots_update(
+						entry->ns->iv_pool_uuid,
+						civ_key->cont_uuid,
+						&civ_ent->iv_snap.snaps[0],
+						civ_ent->iv_snap.snap_cnt);
+			if (rc)
+				return rc;
+		}
 	}
 
 	memcpy(&root_hdl, entry->iv_value.sg_iovs[0].iov_buf, sizeof(root_hdl));
 	d_iov_set(&key_iov, civ_key, sizeof(*civ_key));
 	if (src == NULL) {
 		/* Delete the entry for invalidate */
-		rc = dbtree_delete(root_hdl, BTR_PROBE_EQ, &key_iov, NULL);
+		if (uuid_is_null(civ_key->cont_uuid)) {
```

Comment:
Could you add a comment to explain why it need to delete all entries when cont_uuid is null?

Reply:
sure